app/src/main/java/com/amanoteam/unalix/activities/MainActivity.java

```diff
@@ -1,282 +1,176 @@
 package com.amanoteam.unalix.activities;
 
-import java.util.ArrayList;
-import java.util.List;
-
-import android.app.UiModeManager;
-import android.content.ComponentName;
 import android.content.Intent;
 import android.content.SharedPreferences;
 import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
-import android.content.pm.ResolveInfo;
-import android.content.pm.PackageManager;
-import android.content.res.Configuration;
-import android.net.Uri;
-import android.os.Build;
 import android.os.Bundle;
-import android.os.Parcelable;
 import android.text.TextUtils;
 import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
-import android.view.View.OnClickListener;
 import android.view.View;
+import android.view.View.OnClickListener;
 import android.widget.Toast;
+
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.app.AppCompatDelegate;
 import androidx.appcompat.widget.AppCompatEditText;
-import androidx.appcompat.widget.AppCompatImageButton;
 import androidx.appcompat.widget.Toolbar;
-import androidx.core.content.ContextCompat;
 import androidx.preference.PreferenceManager;
 
-import com.amanoteam.unalix.wrappers.Unalix;
 import com.amanoteam.unalix.R;
-import com.amanoteam.unalix.activities.SettingsActivity;
+import com.amanoteam.unalix.utilities.PackageUtils;
+import com.amanoteam.unalix.wrappers.Unalix;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
 
 public class MainActivity extends AppCompatActivity {
-	
-	private Unalix unalix;
-	
-	private AppCompatEditText urlInput;
-	
-	private PackageManager packageManager;
-	
-	private final ArrayList<ComponentName> excludeTargets = new ArrayList<>();
-	
-	private final ComponentName clearUrlActivity = new ComponentName("com.amanoteam.unalix", "com.amanoteam.unalix.ClearURLActivity");
-	private final ComponentName unshortUrlActivity = new ComponentName("com.amanoteam.unalix", "com.amanoteam.unalix.UnshortURLActivity");
-	
-	private final OnSharedPreferenceChangeListener onSharedPreferenceChangeListener = new SharedPreferences.OnSharedPreferenceChangeListener() {
-		@Override
-		public void onSharedPreferenceChanged(final SharedPreferences settings, final String key) {
-			recreate();
-		}
+
+	private final OnSharedPreferenceChangeListener onSharedPreferenceChangeListener = (settings, key) -> {
+		recreate();
 	};
```

Comment:
I'm not sure whether this `recreate()` is still necessary, but I left it as is for now.